### PR TITLE
add aligned roi introduced in Detectron2

### PR DESCRIPTION
**Pull request by Jerryzcn**
## Description ##
(Brief description on what this PR is about)
add aligned ROIAlign introduced in https://github.com/pytorch/pytorch/pull/23706
improved MaskRCNN mask acc by 0.2~0.3 mAP. However when training with large batch size, warmup needs to be increased.

### CODE REVIEW

**Comment by Jerryzcn** [comment]
@Kh4L

File: src/operator/contrib/mrcnn_mask_target.cu [hunk 1]
```diff
@@ -135,14 +136,21 @@ __device__ void RoIAlignForward(
...
+    T roi_end_w = offset_rois[2] - roi_offset;
+    T roi_end_h = offset_rois[3] - roi_offset;
+
+    T roi_width = roi_end_w - roi_start_w;
+    T roi_height = roi_end_h - roi_start_h;
 
     // Force malformed ROIs to be 1x1
-    T roi_width = max(roi_end_w - roi_start_w, (T)1.);
-    T roi_height = max(roi_end_h - roi_start_h, (T)1.);
+    if (!continuous_coordinate) {  // backward compatiblity
```
Comment (Kh4L) [inline src/operator/contrib/mrcnn_mask_target.cu]: Why we don't also `CHECK_GT` here?

Comment (Jerryzcn) [reply src/operator/contrib/mrcnn_mask_target.cu]: save computation

**Comment by chongruo** [comment]
LGTM. Thanks

**Comment by wkcn** [comment]
Hi @Jerryzcn , could you please add an unittest? Thank you!
